toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -951,79 +951,145 @@ func InstallGrubCfg(installRoot, rootDevice, bootUUID, bootPrefix string, encryp
 	const (
 		assetGrubcfgFile = "/installer/grub2/grub.cfg"
 		grubCfgFile      = "boot/grub2/grub.cfg"
```

Comment:
instead of the toolkit installing a custom-crafted grub.cfg file *and* the grub configuration defaults, this should be *only* installing the grub defaults file and then actually running grub to generate the grub.cfg in the image...

Reply:
Right, but the scope of this PR is only to enable mkconfig generation, not entirely swap the mechanism over to grub2-mkconfig. I.E. we still want the tools to generate the grub.cfg at image build time. That being said, I think that the %post section of grub2 is the ideal place to configure the image (even for first boot) once we do switch over.

Reply:
I think it's dangerous to both claim that grub-mkconfig can be used but also keep generating images that don't actually contain the output of grub-mkconfig. I think we need to switch the image build tooling over to actually using grub-mkconfig at the same time we update grub-mkconfig to generate what the image tooling was previously hardcoding into images.

Reply:
Just confirming some of our offline conversation - the source of `/etc/default/grub` comes from our image generation tooling (i.e., provided as part of the image) and not from the grub package. Is this correct?

Reply:
Yes. The /etc/default/grub packaged with grub2.spec gets stomped if it is present. Otherwise, (if, say, you install or update grub2 on an existing image) %config(noreplace) prevents the tooling-provided /etc/default/grub from being stomped again by the rpm.